docs/choreolib/auto-factory.md

```diff
@@ -1,109 +1,282 @@
 
-# AutoFactory
+# AutoFactory (Java Only)
 
-ChoreoLib provides the `AutoFactory` class as a higher level API to make it easier to create complex auto routines inside your robot code.
+!!! note
+    This API is only available for Java. C++ and Python users should instead utilize the [Trajectory API](./trajectory-api.md).
 
-## Basic Usage
+ChoreoLib provides the `AutoFactory` ([Java](/api/choreolib/java/choreo/auto/AutoFactory.html)) class as a higher level API to make it easier to create complex auto routines inside your robot code.
 
-You can set up the `AutoFactory` by calling `Choreo.createAutoFactory`.
-The `AutoFactory` is recommended to be created in the robot class constructor instead of the drive subsystem.
+```java title="Robot.java"
+public class Robot extends TimedRobot {
+    private final Drive driveSubsystem = new Drive();
+    private final AutoFactory autoFactory;
+
+    public Robot() {
+        autoFactory = new AutoFactory(
+            driveSubsystem::getPose, // A function that returns the current robot pose
+            driveSubsystem::resetOdometry, // A function that resets the current robot pose to the provided Pose2d
+            driveSubsystem::followTrajectory, // The drive subsystem trajectory follower (1)
+            () -> true, // If alliance flipping should be enabled (2)
```

Comment:
With #1025 changing this to a boolean instead of a supplier, this should be updated to match